docs/DeveloperGuide.md

```diff
@@ -1,15 +1,126 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}'
+
+## Getting started
+
+Refer to the user guide [here](https://github.com/AY2223S2-CS2113-F13-3/tp/blob/master/docs/UserGuide.md).
+
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Parser Component
+The component will return the correct command based on what the user inputs into the application.
+
+#### How is the feature implemented?
+
+The Parser component parses the command of the user input and breaks the user input into different parts based on the flags.
+This component also ensures to validate that user input is correct.
+
+The Parser component handles the following methods:
+* Add event
+* Add module
+* List event list
+* Delete event
+* Edit event
+* Find date of week
+* Format checker
+* Revise information using index
+* Revise information using name
+
+The UML diagram below illustrates the flow of how the application adds events:
+![Add Event Class Diagram](UML/Images/addEvent.png)
+
+
+The UML diagram below illustrates the flow of how the application adds modules:
+![Add Event Class Diagram](UML/Images/addModules.png)
+
+#### Why is the feature implemented this way?
+
+This component should be able to guide the user to inputting the correct format of the command to do data validation.
+It should be able to perform basic data validation checks to ensure that the user does not enter any invalid commands.
+This makes life easier to developers in the future if they wish to add new features that requires users to use new commands.
+
+### Storage Component
+API: `Storage.java`
+
+The storage component reads and writes user data from a local save in the form of a `.json` file.
+The Storage component:
+* Serializes and deserializes user data into a `.json` file format through the use of the Gson library
+* Loads data from `NusMods.json` into a HashMap for use by other classes.
+* Inherits from both `EventListStorage` and `NusModuleLoader`, and can be treated as either one.
+* Saves and loads information from the local hard disk
+* Depends on some classes (the `storage` component saves and retrieves objects)
+
+The class diagram below illustrates the structure of the storage package
+
+![Storage Class Diagram](UML/Images/StorageClass.png)
+
+#### How the feature is implemented:
+
+##### Load Events
+
+![Load Events Sequence Diagram](UML/Images/loadEvents.png)
+
+When the application starts up, the storage loadEvents() function will be called to load contents in the save file. 
+
+##### Load Modules
+![Load Modules Sequence Diagram](UML/Images/LoadModules.png)
+
+When any component requires reading the NUS module files, the loadModules() is called.
+
+##### Save Events
+
+![Save To File Sequence Diagram](UML/Images/SaveToFile.png)
```

Comment:
Perhaps the activation bar can be a single one rather than split as the object is created once and called again later?
![image](https://user-images.githubusercontent.com/88631870/228730768-34f1ddf1-6741-473c-b90e-75d8bccd3ac5.png)